tests/png/bug00295.c

```diff
@@ -0,0 +1,39 @@
+/**
+ * Regression test for <https://github.com/libgd/libgd/issues/295>.
+ *
+ * We're testing that reading a PNG which might trigger a libpng warning
+ * does not trigger a GD_ERROR.
+ */
+
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdtest.h"
+#include <string.h>
+
+
+static void error_handler(int priority, const char *format, va_list args)
+{
+    char msg[256];
+
+    vsprintf(msg, format, args);
+    gdTestAssertMsg(priority != GD_ERROR,
+                    "unexpected GD_ERROR: %s\n", msg);
+}
+
+
+int main()
+{
+    gdImagePtr im;
+    FILE *fp;
+
+    gdSetErrorMethod(error_handler);
+    fp = gdTestFileOpen2("png", "bug00295.png");
+
+    im = gdImageCreateFromPng(fp);
+
+    fclose(fp);
+    gdImageDestroy(im);
+
+    return gdNumFailures();

```

Comment:
to be clear, i'm not saying you must do this ... it'd just be a nice to have since the current test doesn't actually hard verify that we set the error handler.  if we broke things in the future, this test case wouldn't help catch regressions.  i.e. i'm pretty sure it'll "pass" if you build it against a version of gd that doesn't have your fixes here.